oauth2client/client.py

```diff
@@ -837,6 +838,9 @@ def _generate_refresh_request_headers(self):
         if self.user_agent is not None:
             headers['user-agent'] = self.user_agent
 
+        if _METADATA_V1_URL_PREFIX in self.token_uri:
+            headers[_METADATA_FLAVOR_HEADER] = _DESIRED_METADATA_FLAVOR
```

Comment:
Looking at some of our code, it seems that we are directly instantiating an oauth2client.client.OAuth2Credentials object with a metadata URL.  How does the AppAssertionCredentials come into play at that point?